### PR TITLE
Fix: Import Kit Issues - other plugins interference in the Import process [ED-12524]

### PR DESCRIPTION
**Pull request by Omerisra6**

### CODE REVIEW

File: app/modules/import-export/module.php [hunk 1]
```diff
@@ -867,4 +882,35 @@ private function get_elementor_editor_page_url( $page_id ) {
...
+	/**
+	 * @param string $class
+	 *
+	 * @return bool
+	 */
+	private function is_third_party_plugin_class( $class ) {
+		$allowed_classes = [
+			'Elementor',
+			'ElementorPro',
+			'WP',
```
Comment (mark-elementor) [inline app/modules/import-export/module.php:895]:
```suggestion
			'Elementor\\',
			'ElementorPro\\',
			'WP_',
			'wp_',
```

File: app/modules/import-export/module.php [hunk 2]
```diff
@@ -541,7 +552,11 @@ private function maybe_handle_ajax() {
 			}
 		} catch ( \Error $e ) {
 			if ( isset( $this->import ) ) {
-				$this->import->finalize_import_session_option();
+				$is_third_party_plugin_error = $this->is_third_party_plugin_class( $e->getTrace()[0]['class'] );
+
+				$is_third_party_plugin_error ?
+					wp_send_json_error( self::THIRD_PARTY_PLUGIN_ERROR, 500 ) :
+					$this->import->finalize_import_session_option();
```
Comment (mark-elementor) [inline app/modules/import-export/module.php:559]:
```suggestion
				if ( ! $this->is_native_class( $e->getTrace()[0]['class'] ) ) {
					wp_send_json_error( self::THIRD_PARTY_PLUGIN_ERROR, 500 );
				}

				$this->import->finalize_import_session_option() :
```

Comment (mark-elementor) [reply app/modules/import-export/module.php:559]: Also, I'm not sure this is the right place to add the check. As basically we want to know which runner failed with the error IMO. Let's talk about it?

File: app/modules/import-export/assets/js/shared/process-failed-dialog/process-failed-dialog.js [hunk 3]
```diff
@@ -60,6 +60,14 @@ const messagesContent = {
 		title: __( 'Couldn’t install the Kit', 'elementor' ),
 		text: __( 'The Kit includes plugins you don’t have permission to install. Contact your site admin to change your permissions.', 'elementor' ),
 	},
+	'no-domdocument': {
```
[the remark this answers is not in the document]
Comment (mark-elementor) [reply app/modules/import-export/assets/js/shared/process-failed-dialog/process-failed-dialog.js:63]: Also, should not be part of this PR.

File: app/modules/import-export/module.php (hunk 1 above)
Comment (Omerisra6) [reply app/modules/import-export/module.php:898]: Done.

File: app/modules/import-export/module.php [hunk 4]
```diff
@@ -867,4 +882,35 @@ private function get_elementor_editor_page_url( $page_id ) {
...
+		return ! $this->str_starts_with_any( $class, $allowed_classes );
+	}
+
+	/**
+	 * @param $haystack
+	 * @param array $needles
+	 *
+	 * @return bool
+	 */
+	private function str_starts_with_any( $haystack, array $needles ) {
```
[the remark this answers is not in the document]
Comment (Omerisra6) [reply app/modules/import-export/module.php:910]: Done.

File: app/modules/import-export/module.php [hunk 5]
```diff
@@ -867,4 +882,35 @@ private function get_elementor_editor_page_url( $page_id ) {
 
 		return $document ? $document->get_edit_url() : '';
 	}
+
+	/**
+	 * @param string $class
+	 *
+	 * @return bool
+	 */
+	private function is_third_party_plugin_class( $class ) {
```
[the remark this answers is not in the document]
Comment (Omerisra6) [reply app/modules/import-export/module.php:894]: Done

File: app/modules/import-export/assets/js/shared/process-failed-dialog/process-failed-dialog.js [hunk 6]
```diff
@@ -60,6 +60,14 @@ const messagesContent = {
...
+		title: __( 'Couldn’t install the Kit', 'elementor' ),
+		text: __( 'You must have DOMDocument installed', 'elementor' ),
+	},
+	'third-party-plugin-error': {
```
[the remark this answers is not in the document]
Comment (Omerisra6) [reply app/modules/import-export/assets/js/shared/process-failed-dialog/process-failed-dialog.js:67]: Done.

[the remark this answers is not in the document]
Comment (mark-elementor) [reply app/modules/import-export/assets/js/shared/process-failed-dialog/process-failed-dialog.js:67]: Don't see it.

File: app/modules/import-export/assets/js/shared/process-failed-dialog/process-failed-dialog.js [hunk 7]
```diff
@@ -60,6 +60,14 @@ const messagesContent = {
 		title: __( 'Couldn’t install the Kit', 'elementor' ),
 		text: __( 'The Kit includes plugins you don’t have permission to install. Contact your site admin to change your permissions.', 'elementor' ),
 	},
+	'domdocument-missing': {
+		title: __( 'Couldn’t install the Kit', 'elementor' ),
+		text: __( 'You must have DOMDocument installed', 'elementor' ),
+	},
```
Comment (mark-elementor) [inline app/modules/import-export/assets/js/shared/process-failed-dialog/process-failed-dialog.js:66]: You have a mixup of two PRs. Please take these changes out of this one.
```suggestion
```

File: app/modules/import-export/module.php [hunk 8]
```diff
@@ -43,6 +43,10 @@ class Module extends BaseModule {
 
 	const NO_WRITE_PERMISSIONS_KEY = 'no-write-permissions';
 
+	const DOMDOCUMENT_MISSING = 'domdocument-missing';
+
```
Comment (mark-elementor) [inline app/modules/import-export/module.php:47]: You have a mixup of two PRs. Please take these changes out of this one.
```suggestion
```

File: app/modules/import-export/module.php [hunk 9]
```diff
@@ -331,6 +335,7 @@ public function upload_kit( $file, $referrer ) {
 	 */
 	public function import_kit( string $path, array $settings, bool $split_to_chunks = false ): array {
 		$this->ensure_writing_permissions();
+		$this->ensure_DOMDocument_exists();
```
Comment (mark-elementor) [inline app/modules/import-export/module.php:338]: You have a mixup of two PRs. Please take these changes out of this one.
```suggestion
```

File: app/modules/import-export/module.php [hunk 10]
```diff
@@ -489,6 +494,12 @@ private function ensure_writing_permissions() {
 		}
 	}
 
+	private function ensure_DOMDocument_exists() {
+		if ( ! class_exists( 'DOMDocument' ) ) {
+			throw new \Error( self::DOMDOCUMENT_MISSING );
+		}
+	}
+
```
Comment (mark-elementor) [inline app/modules/import-export/module.php:502]: You have a mixup of two PRs. Please take these changes out of this one.
```suggestion
```

File: app/modules/import-export/assets/js/shared/process-failed-dialog/process-failed-dialog.js [hunk 11]
```diff
@@ -60,6 +60,14 @@ const messagesContent = {
...
+	'third-party-plugin-error': {
+		title: __( 'Couldn’t install the Kit', 'elementor' ),
+		text: __( 'Other plugins ', 'elementor' ),
```
Comment (mark-elementor) [inline app/modules/import-export/assets/js/shared/process-failed-dialog/process-failed-dialog.js:69]: Don't forget to update the text before merging :)

File: app/modules/import-export/assets/js/shared/process-failed-dialog/process-failed-dialog.js (hunk 7 above)
Comment (Omerisra6) [reply app/modules/import-export/assets/js/shared/process-failed-dialog/process-failed-dialog.js:66]: This branch s coming out from the other branch...
Should i rebase him?

File: app/modules/import-export/module.php (hunk 9 above)
Comment (Omerisra6) [reply app/modules/import-export/module.php:338]: Done.

File: app/modules/import-export/module.php (hunk 10 above)
Comment (Omerisra6) [reply app/modules/import-export/module.php:502]: Done.

**Pull request closed without merging**